internal/parser/parser.go

```diff
@@ -1061,50 +1062,122 @@ func (p *parser) parsePropertyList() []*ast.Property {
 		case token.STRING:
 			param = p.parseStringProperty()
 		default:
-			// TODO(jsternberg): BadExpression.
-			p.consume()
-			continue
+			param = p.parseInvalidProperty()
 		}
 		params = append(params, param)
-		if _, tok, _ := p.peek(); tok == token.COMMA {
-			p.consume()
+
+		if p.more() {
+			if _, tok, lit := p.peek(); tok != token.COMMA {
+				perrs = append(perrs, ast.Error{
+					Msg: fmt.Sprintf("expected comma in property list, got %s (%q)", tok, lit),
+				})
+			} else {
+				p.consume()
+			}
 		}
 	}
+	p.errs = append(p.errs, perrs...)
 	return params
 }
 
 func (p *parser) parseStringProperty() *ast.Property {
 	key := p.parseStringLiteral()
 	p.expect(token.COLON)
-	val := p.parseExpression()
+	val := p.parsePropertyValue()
+	var endPos ast.Position
+	if val != nil {
```

Comment:
I understand why this is here, but can you use the current pattern of calling `locEnd` on only the value so this is consistent with the other parts of the code?

Reply:
Okay, I'll do that.  The location info will be the zero value for properties with no value set for now.  I filed an issue for  your idea of `loc(nodes ...ast.Node)`:
https://github.com/influxdata/flux/issues/1237